umake/ui/cli/__init__.py

```diff
@@ -157,6 +157,60 @@ def mangle_args_for_default_framework(args):
     return result_args
 
 
+def list_frameworks(listing, detailled=True):
+    if listing == "all" or listing == "available":
+        # Sort the categories to prevent a random list at each new program execution
+        for category in sorted(BaseCategory.categories.values(), key=lambda cat: cat.prog_name):
+            # The category must have at least one framework
+            if len(category.frameworks) == 0:
+                continue
+
+            if detailled:
+                print("{}: {}".format(category.prog_name, category.description))
+            else:
+                print("{}:".format(category.prog_name))
+
+            # Sort the frameworks to prevent a random list at each new program execution
+            for framework in sorted(category.frameworks.values(), key=lambda fram: fram.prog_name):
+                # Framework doesn't be removal
+                if framework.only_for_removal:
```

Comment:
So, as shown above, we return frameworks marked as only_for_removal. We will only print it though when the framework is installed and will only (deprecated, only for removal) for instance.